### ISSUE TITLE
grunt concurrent make the job crash

### ISSUE THREAD

**Issue by yarmand** [Microsoft/vsts-agent#1097]
## Agent version and platform
Version of your agent? 2.119.1

OS of the machine running the agent? Linux...

## VSTS type and version
VisualStudio.com

If VisualStudio.com, what is your account name? http://yammer.visualstudio.com

## What's not working?
Script job dies when using grunt_concurrent resulting into a failing build.
I run the agent inside a docker container, I tried to run the grunt command directly in the container and everything is fine. Only when I run within the agent grunt fails when reaching concurrent tasks.
I narrow down grunt concurrent to use [grunt.util.spawn](https://github.com/sindresorhus/grunt-concurrent/blob/master/tasks/concurrent.js#L37).

I have tried forking and spawning in different language within the agent on linux and it works well.

I tried building with grunt_concurrent on windows and do not see the issue either.

## Agent and Worker's diag log
```
2017-07-26T21:20:18.0671540Z Running "concurrent:build" (concurrent) task
2017-07-26T21:20:18.0683780Z [D] Task source: /vsts/agent/_work/1/s/node_modules/grunt-concurrent/tasks/concurrent.js
2017-07-26T21:20:18.0696710Z >> Warning: There are more tasks than your concurrency limit. After this limit
2017-07-26T21:20:18.0712440Z >> is reached no further tasks will be run until the current tasks are
2017-07-26T21:20:18.0725430Z >> completed. You can adjust the limit in the concurrent task options
2017-07-26T21:20:22.6848800Z 
2017-07-26T21:20:22.6886330Z     Running "webpack:staging_canary" (webpack) task
2017-07-26T21:20:22.6898440Z     [D] Task source: /vsts/agent/_work/1/s/node_modules/grunt-webpack/tasks/webpack.js
2017-07-26T21:20:22.6910230Z Warning:  Use --force to continue.
2017-07-26T21:20:22.6921630Z Error
2017-07-26T21:20:22.6933530Z     at callDone (/vsts/agent/_work/1/s/node_modules/grunt-legacy-util/index.js:159:52)
2017-07-26T21:20:22.6954660Z     at ChildProcess.<anonymous> (/vsts/agent/_work/1/s/node_modules/grunt-legacy-util/index.js:198:5)
2017-07-26T21:20:22.6969480Z     at emitTwo (events.js:125:13)
2017-07-26T21:20:22.6982910Z     at ChildProcess.emit (events.js:213:7)
2017-07-26T21:20:22.6996680Z     at maybeClose (internal/child_process.js:921:16)
2017-07-26T21:20:22.7011230Z     at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
2017-07-26T21:20:22.7018390Z 
2017-07-26T21:20:22.7031810Z Aborted due to warnings.
```

**Comment by bryanmacfarlane** [comment]
This is not the agent.  The agent just runs the task which runs grunt.  That error is coming from grunt-webpack.  The agent knows nothing about grunt

**Comment by yarmand** [comment]
@bryanmacfarlane can you explain me why this is working in the same container if I launch the grunt task manually, but do not work when the grunt task is launched by the agent ?
The only changing part between these 2 attempts is the agent nesting the grunt process.

**Comment by yarmand** [comment]
Nevermind. IT was a memory capacity issue with my container. running the agent just throw it over the fence. Increasing memory on the container fixed the issue.

**Comment by bryanmacfarlane** [comment]
Cool.  I was going to suggest something environmental (OS, capacity etc...).  Glad you found it.  For example, we saw issues in npm that worked on dev machines but not on azure build machines.  Ended up being I/O constraints highlighting a race in npm.

The other differences are security context - account agent runs as, profiles etc...

The agent simply execs the process and we show it in the log.

Glad you found your issue.  Hopefully your issue and stack error will help someone else :)
